src/Tasks/Monitor/BackupDestinationStatus.php

```diff
@@ -2,158 +2,64 @@
 
 namespace Spatie\Backup\Tasks\Monitor;
 
-use Exception;
-use Carbon\Carbon;
-use Spatie\Backup\Helpers\Format;
 use Spatie\Backup\BackupDestination\BackupDestination;
+use Spatie\Backup\Tasks\Monitor\HealthChecks\IsReachable;
 
 class BackupDestinationStatus
 {
     /** @var \Spatie\Backup\BackupDestination\BackupDestination */
     protected $backupDestination;
 
-    /** @var int */
-    protected $maximumAgeOfNewestBackupInDays = 1;
+    /** @var array */
+    protected $healthChecks;
 
-    /** @var int */
-    protected $maximumStorageUsageInMegabytes = 5000;
+    /** @var HealthCheckFailure|null */
+    protected $failedHealthCheck;
 
-    /** @var string */
-    protected $diskName;
-
-    /** @var bool */
-    protected $reachable;
-
-    public function __construct(BackupDestination $backupDestination, string $diskName)
+    public function __construct(BackupDestination $backupDestination, array $healthChecks = [])
     {
         $this->backupDestination = $backupDestination;
-        $this->diskName = $diskName;
-
-        $this->reachable = $this->backupDestination->isReachable();
-    }
-
-    public function setMaximumAgeOfNewestBackupInDays(int $days): self
-    {
-        $this->maximumAgeOfNewestBackupInDays = $days;
-
-        return $this;
+        $this->healthChecks = $healthChecks;
     }
 
-    public function maximumAgeOfNewestBackupInDays(): int
-    {
-        return $this->maximumAgeOfNewestBackupInDays;
-    }
-
-    public function setMaximumStorageUsageInMegabytes(float $megabytes): self
-    {
-        $this->maximumStorageUsageInMegabytes = $megabytes;
-
-        return $this;
-    }
-
-    public function backupName(): string
-    {
-        return $this->backupDestination->backupName();
-    }
-
-    public function diskName(): string
-    {
-        return $this->diskName;
-    }
-
-    public function amountOfBackups(): int
-    {
-        return $this->backupDestination->backups()->count();
-    }
-
-    public function dateOfNewestBackup(): ?Carbon
+    public function backupDestination(): BackupDestination
     {
-        $newestBackup = $this->backupDestination->newestBackup();
-
-        if (is_null($newestBackup)) {
-            return null;
-        }
-
-        return $newestBackup->date();
+        return $this->backupDestination;
     }
 
-    public function newestBackupIsTooOld(): bool
+    public function check(HealthCheck $check)
     {
-        if (! count($this->backupDestination->backups())) {
-            return true;
+        try {
+            $check->handle($this->backupDestination());
+        } catch (\Exception $exception) {
+            return new HealthCheckFailure($check, $exception);
```

Comment:
Create a static methods on that exception, so we don't need to new it up.

More info: http://rosstuck.com/formatting-exception-messages

Reply:
In fact this is not an exception itself but a wrapper around the exception with a bunch of help methods to interact with the health check and the failure.

I'm not sure how to apply the tactics in this case?